test/jdk/java/util/concurrent/CompletableFuture/CompletableFutureOrTimeoutExceptionallyTest.java

```diff
@@ -30,39 +30,24 @@
  */
 
 import java.time.Duration;
-import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ForkJoinPool;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ScheduledThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
 import org.junit.jupiter.api.Test;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 class CompletableFutureOrTimeoutExceptionallyTest {
-    static final BlockingQueue<Runnable> delayerQueue;
-    static {
-        try {
-            var delayerClass = Class.forName("java.util.concurrent.CompletableFuture$Delayer",
-                                             true,
-                                             CompletableFuture.class.getClassLoader());
-            var delayerField = delayerClass.getDeclaredField("delayer");
-            delayerField.setAccessible(true);
-            delayerQueue = ((ScheduledThreadPoolExecutor)delayerField.get(null)).getQueue();
-        } catch (Throwable t) {
-            throw new ExceptionInInitializerError(t);
-        }
-    }
-
+    // updated February 2025 to adapt to CompletableFuture DelayScheduler changes
     /**
      * Test that orTimeout task is cancelled if the CompletableFuture is completed Exceptionally
      */
     @Test
     void testOrTimeoutWithCompleteExceptionallyDoesNotLeak() throws InterruptedException {
-        assertTrue(delayerQueue.peek() == null);
+        ForkJoinPool delayer = ForkJoinPool.commonPool();
         var future = new CompletableFuture<>().orTimeout(12, TimeUnit.HOURS);
-        assertTrue(delayerQueue.peek() != null);
         future.completeExceptionally(new RuntimeException("This is fine"));
-        while (delayerQueue.peek() != null) {
+        while (delayer.getDelayedTaskCount() != 0) {
```

Comment:
Might be good to start the test case asserting that the delayed task count is 0 (so there's no interference).

Reply:
Thanks, done.